### ISSUE TITLE
Support unnamed globals, e.g. @""

### ISSUE THREAD

**Issue by mewmew** [llir/llvm#118]
From https://github.com/llir/llvm/issues/111#issuecomment-562429501

## Assign global IDs and support unnamed globals

`llir/llvm` currently rejects `llvm/test/CodeGen/X86/extractps.ll` as it contains multiple function definitions with the name `@""`, which causes a name collision.

However, `opt` accepts this input and treats `@""` as unnamed, and assigns global IDs to these unamed global identifiers.

Using `opt -S -o a.ll llvm/test/CodeGen/X86/extractps.ll` produces:
```llvm
@0 = external global float

define void @1() { ; <--- given ID 1 from unnamed ""
...
}
define void @2() { ; <--- given ID 2 from unnamed ""
...
}
```

Extract of llvm/test/CodeGen/X86/extractps.ll:
```llvm
@0 = external global float

define void @""() {
   load float, float* @0, align 16
   insertelement <4 x float> undef, float %1, i32 0
   call <4 x float> @llvm.x86.sse.rsqrt.ss( <4 x float> %2 )
   extractelement <4 x float> %3, i32 0
   store float %4, float* @0, align 16
   ret void
}
define void @""() {
   load float, float* @0, align 16
   insertelement <4 x float> undef, float %1, i32 1
   call <4 x float> @llvm.x86.sse.rsqrt.ss( <4 x float> %2 )
   extractelement <4 x float> %3, i32 1
   store float %4, float* @0, align 16
   ret void
}

declare <4 x float> @llvm.x86.sse.rsqrt.ss(<4 x float>)
```

Only one test case exists which has multiple unnamed global definitions, namely:

* [llvm/test/CodeGen/X86/extractps.ll](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/CodeGen/X86/extractps.ll)
   - global identifier "@0" already present; prev `@0 = external global float, align 16`, new `define internal void @""() nounwind {

**Comment by dannypsnl** [comment]
I create a little test for this:

```go
import (
	"fmt"
	"testing"

	"github.com/llir/llvm/asm"
)

func TestASMParse(t *testing.T) {
	// Parse the LLVM IR assembly file `foo.ll`.
	m, err := asm.ParseFile("extractps.ll")
	if err != nil {
		t.Fatalf("%+v", err)
	}
	fmt.Printf("%s\n", m.String())
}
```

to fix this problem I have two ideas and related to #75 
1. disable the panic when a `GlobalIdent.GlobalName` is `""`, in the case it would generate three `@0` global name, and we can manually use **irutil** to fix these duplicated. The benefit was the approach can be quickly done. But didn't quite correctly mapping to the LLVM concept, LLVM consider `""` global name as undefined so rename it later. If we choose to keep behavior as #75 then this is not a big problem.
2. Follow LLVM concept treat `""` as undefined and handle the naming for them. This would take a longer time to analyze and implement. It basically is not how current implementation logic, would need to reconsider the order of operations about global naming, cannot simply give ID for unnamed globals.

**Comment by mewmew** [comment]
Thanks for investigating this @dannypsnl. While approach 1 may be quicker to implement, I think approach 2 will lead to a more solid foundation to build upon. Similarly to how we treat local variables as unnamed, and later assign IDs to them using `Func.AssignIDs` I envision we would do something similar for global variables.

**Comment by mewmew** [comment]
Fixes by @dannypsnl in #127.
